### PR TITLE
Action Button

### PR DESCRIPTION
**Pull request by SarahPappas**
Create a generic action button for inputs or other actions.

### CODE REVIEW

File: src/components/UploadPhoto/UploadPhoto.tsx [hunk 1]
```diff
@@ -25,17 +26,13 @@ export const UploadPhoto: React.FC<UploadPhotoProps> = ({setUploadedFileData}) =
             reader.readAsDataURL(file);
         }
     };
+
+    const uploadImageButton = { text: 'UPLOAD PHOTO', to: '' };
```
Comment (msdrey) [inline src/components/UploadPhoto/UploadPhoto.tsx]: Do action buttons need a "to" field? Should we have different types for NavButtons and ActionButtons?  It feels weird to have 1 type but 2 classes. Or maybe we can we make 1 type and 1 class (with an optional "to", "input" and so forth)? I wish we were in Java and could have a non-instanciable interface or abstract class for Button with different classes of buttons inheriting from that. I'm not sure if that's possible in react though.

**Review by msdrey — COMMENTED** [review]
I like how the input is integrated in our action button, it makes things really clean in the tsx pages. What were you thinking about the addPhoto page, where there is a button that is displayed following a condition? should we integrate that functionality in the NavButton? or integrate navbuttons into a more generic class of buttons?

File: src/components/UploadPhoto/UploadPhoto.tsx (hunk 1 above)
Comment (SarahPappas) [reply src/components/UploadPhoto/UploadPhoto.tsx]: No, they don't need a to field. I was thinking about the Button type more as an interface. I think maybe we should rename the button type to navbutton and not use it in the action button.

File: src/components/NavButton/NavButton.tsx [hunk 2]
```diff
@@ -3,7 +3,7 @@ import {Link, Route} from 'react-router-dom';
 import './NavButton.css';
 
 interface NavButtonProps {
-    button: Button;
+    button: NavButton;
```
Comment (msdrey) [inline src/components/NavButton/NavButton.tsx:7]: I might be nitpicky, and feel free to dismiss this... we now have 2 "sibling" button classes, NavButton and ActionButton, that take in their props differently: in NavButton we group them into a type, in ActionButton we take them in separately. Do we want to uniformize the way button are created for a more coherent coding experience?

File: src/components/ActionButton/ActionButton.tsx [hunk 3]
```diff
@@ -2,15 +2,15 @@ import React from 'react';
...
 }
 
-export const ActionButton: React.FC<ActionButtonProps> = ({ button, action, input }) => {    
+export const ActionButton: React.FC<ActionButtonProps> = ({ text, action, input }) => {    
     const buttonLabel = <> 
         <div className='label'> 
-            {button.text}
+            {text}
         </div>
     </>;
```
Comment (msdrey) [inline src/components/ActionButton/ActionButton.tsx:11]: extra space in line 29

File: src/components/Button/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+interface NavButtonProps {
```
Comment (msdrey) [inline src/components/Button/Button.tsx]: The interface should probably be named ButtonProps

File: src/components/InputButton/InputButton.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Button } from 'components/Button/Button';
+import './InputButton.css';
+
+interface InputButtonProps {
+    button: ActionButton;
+    input?: Input;
```
Comment (msdrey) [inline src/components/InputButton/InputButton.tsx]: Maybe the input should be required for inputbuttons?

File: src/components/Button/Button.tsx (hunk 4 above)
Comment (SarahPappas) [reply src/components/Button/Button.tsx]: yep, I will update thanks!

File: src/components/InputButton/InputButton.tsx (hunk 5 above)
Comment (SarahPappas) [reply src/components/InputButton/InputButton.tsx]: yes!

**Pull request merged**